test_launch_ros/test/test_launch_ros/actions/test_node.py

```diff
@@ -358,6 +362,111 @@ def test_launch_node_with_invalid_parameter_dicts(self):
                 },
             }])
 
+    def _add_secure_flag(self, actions: List) -> List:
+        return [DeclareLaunchArgument('__secure', default_value=['true'])] + actions
+
+    def _add_secure_no_create_flag(self, actions: List) -> List:
+        return [
+            DeclareLaunchArgument('__no_create_keystore', default_value=['true'])
+        ] + self._add_secure_flag(actions)
+
+    def _assert_has_enclave(self, node, node_name):
+        assert any('--enclave' in cmd[0].describe() for cmd in node.cmd)
+        assert node._Node__enclave == node_name
+
+    @patch('launch_ros.actions.node.nodl', autospec=True)
+    def test_launch_node_transient_keystore(self, nodl_mock):
+        nodl_mock.get_node_by_executable.return_value.name = 'foo'
+        with tempfile.TemporaryDirectory() as tmpdir:
+            with patch('launch_ros.actions.node.TemporaryDirectory') as tempdir_mock:
+                tempdir_mock.return_value.name = tmpdir
+                node_fixture = self._create_node()
+                actions = self._add_secure_flag([node_fixture])
+                self._assert_launch_no_errors(actions)
+                self._assert_has_enclave(node_fixture, '/my_ns/my_node')
+
+                assert is_valid_keystore(tmpdir)
+
+    @patch('launch_ros.actions.node.nodl', autospec=True)
+    def test_launch_node_transient_keystore_no_create(self, nodl_mock):
+        nodl_mock.get_node_by_executable.return_value.name = 'foo'
+        with tempfile.TemporaryDirectory() as tmpdir:
+            with patch('launch_ros.actions.node.TemporaryDirectory') as tempdir_mock:
+                tempdir_mock.return_value.name = tmpdir
+                # Test disabling keystore creation
+                node_fixture = self._create_node()
+                actions = self._add_secure_no_create_flag([node_fixture])
+                self._assert_launch_errors(actions)
```

Comment:
I have not checked that above, is the error message meaningful here? As in a 'you can't use that flag in that case' kind of msg.

Reply:
Added some specific exceptions now, on lines 473-438